src/al-sap/al_service_utils.cpp

```diff
@@ -64,3 +64,21 @@ bool areMacsEqual(const std::array<uint8_t, 6> &first, const std::array<uint8_t,
 {
     return std::equal(begin(first), end(first), begin(second));
 }
+
+std::vector<unsigned char> convert_u32_into_bytes(uint32_t number)
+{
+    return std::vector<unsigned char>{static_cast<unsigned char>(number >> 24),
+                                      static_cast<unsigned char>(number >> 16),
+                                      static_cast<unsigned char>(number >> 8),
+                                      static_cast<unsigned char>(number >> 0)};
+}
+
+uint32_t convert_bytes_into_u32(const std::vector<unsigned char>& bytes)
+{
+    return static_cast<uint32_t>((bytes[0] << 24) | (bytes[1] << 16) | (bytes[2] << 8) | (bytes[3] << 0));
+}
```

Comment:
Similarly: 
(This also adds a needed bounds check)
```suggestion
uint32_t convert_bytes_into_u32(const std::vector<unsigned char>& bytes)
{
    if (bytes.size() != 4) return 0;
    uint32_t value;
    memcpy(&value, bytes.data(), sizeof(uint32_t));
    return value;
}
```

Reply:
I can add a bounds check if required